drivers/docker/utils.go

```diff
@@ -21,31 +21,41 @@ import (
 	"github.com/docker/docker/registry"
 )
 
+var regexForDockerImages = `^(?:(?P<registry>[^/]+)/)?(?P<repository>[^:@]+)(?::(?P<tag>[^@]+))?(?:@(?P<digest>sha256:[a-fA-F0-9]{64}))?$`
```

Comment:
We already have https://pkg.go.dev/github.com/distribution/reference as a dependency. Maybe we should use that official upstream instead of our own regex (as awesome as this regex is :grin: )? We can probably even ditch the test in that case.

Reply:
Done!